### PR TITLE
Add Pytest

### PR DESCRIPTION
**Pull request by rodrigodelazcano**
# Description
This PR adds the following:

- Remove compatibility with python 3.6 due to python foundation not supporting it any more and dependency issues when using entrypoint plugins (https://github.com/Farama-Foundation/gym-minigrid/pull/205#issue-1326438686)
- Fully implemented the `Renderer` Gym API with `render_mode`. This won't break other libraries since the old rendering API can still be used.
-  Moved `benchmark.py` and `manual_control.py` to `gym_minigrid\` directory.
- Add Pytest to the Github CI. Pytests are organized in two files `tests\test_envs.py` and `tests\test_wrappers.py`. These have a combination of tests previously found in `run_tests.py` and `test_interactive_mode.py`, the `env_checker` test from Gym, and new tests. The division of the pytest functions are as follow:

1. **From old `run_tests.py` and `test_interactive_mode.py`**: `test_envs.py:test_agent_sees_method()`, `test_envs.py:old_run_test()`, `test_envs.py:test_interactive_mode()`, `test_wrappers.py:test_dict_observation_space_wrapper()`, `test_wrappers.py:test_main_wrappers()`, `test_wrappers.py:test_observation_space_wrappers()`, `test_wrappers.py:test_agent_sees_method()`

2. **From Gym**: `test_envs.py:test_env()`, `test_envs.py:test_env_determinism_rollout()`, `test_envs.py:test_render_modes()`

3. **New tests**: `test_wrappers.py:test_reseed_wrapper()`, `test_wrappers.py:test_state_bonus_wrapper()`, `test_wrappers.py:test_action_bonus_wrapper()`

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Screenshots
Please attach before and after screenshots of the change if applicable.

# Checklist:

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `pre-commit run --all-files` (see `CONTRIBUTING.md` instructions to set it up)
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: gym_minigrid/minigrid.py [hunk 1]
```diff
@@ -1242,17 +1258,11 @@ def get_obs_render(self, obs, tile_size=TILE_PIXELS // 2):
 
         return img
 
-    def render(self, mode="human", close=False, highlight=True, tile_size=TILE_PIXELS):
+    def _render(self, mode="human", highlight=True, tile_size=TILE_PIXELS):
```
Comment (pseudo-rnd-thoughts) [inline gym_minigrid/minigrid.py:1261]: Add `close` that will be raise an exception telling the user to use `.close()`

File: gym_minigrid/minigrid.py [hunk 2]
```diff
@@ -1302,8 +1312,14 @@ def render(self, mode="human", close=False, highlight=True, tile_size=TILE_PIXEL
         if mode == "human":
             self.window.set_caption(self.mission)
             self.window.show_img(img)
+        else:
+            return img
 
-        return img
+    def render(self, mode="human", highlight=True, tile_size=TILE_PIXELS):
```
Comment (pseudo-rnd-thoughts) [inline gym_minigrid/minigrid.py:1318]: Add `close` that will be raise an exception telling the user to use `.close()`

File: tests/test_wrappers.py [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+@pytest.mark.parametrize(
+    "env_spec", all_testing_env_specs, ids=[spec.id for spec in all_testing_env_specs]
+)
+def test_main_wrappers(wrapper, env_spec):
+    env = env_spec.make()
+    env = wrapper(env)
+    for _ in range(10):
+        env.reset()
+        env.step(0)
+        env.close()
```
Comment (pseudo-rnd-thoughts) [inline tests/test_wrappers.py:161]: Deindent

File: tests/utils.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
+"""Finds all the specs that we can test with"""
+from typing import Optional
+
+import gym
+import numpy as np
+from gym import logger
+from gym.envs.registration import EnvSpec
+
+
+def try_make_env(env_spec: EnvSpec) -> Optional[gym.Env]:
```
Comment (pseudo-rnd-thoughts) [inline tests/utils.py:10]: It shouldn't need this if we only try to import environments in minigrid

File: tests/utils.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    """Tries to make the environment showing if it is possible. Warning the environments have no wrappers, including time limit and order enforcing."""
+    try:
+        return env_spec.make(disable_env_checker=True).unwrapped
+    except ImportError as e:
+        logger.warn(f"Not testing {env_spec.id} due to error: {e}")
+        return None
+
+
+# Tries to make all gym_minigrid environment to test with
+all_testing_initialised_envs = list(
```
Comment (pseudo-rnd-thoughts) [inline tests/utils.py:20]: As all environments should be valid and non-None so we can just use
```
[env_spec.make(disable_env_checker=True).unwrapped
for env_spec in gym.envs.registry.values()
 if env_spec.entry_point.startswith("gym_minigrid.envs")
]

```

**Review by pseudo-rnd-thoughts — CHANGES_REQUESTED** [review]
Generally looks good. 
In the build information, pytest raises a warning that None should not be used in `pytest.warns()` or `pytest.raises()`

**Comment by rodrigodelazcano** [comment]
I updated the PR with the changes. Also fixed the deprecated None in `pytest.warn()`

**Comment by pseudo-rnd-thoughts** [comment]
LGTM

**Pull request merged**